src/screens/Send/SendScreenHeader.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { getFocusedRouteNameFromRoute } from '@react-navigation/native'
+import { StackHeaderProps } from '@react-navigation/stack'
+import { ChevronLeft } from 'lucide-react-native'
+import { useEffect, useState } from 'react'
+import { PressableProps } from 'react-native'
+import { Bar as ProgressBar } from 'react-native-progress'
+import styled, { useTheme } from 'styled-components/native'
+
+import AppText from '~/components/AppText'
+import BottomModalHeader from '~/components/headers/BottomModalHeader'
+import { ScreenSection } from '~/components/layout/Screen'
+
+const sendSteps = ['DestinationScreen', 'OriginScreen', 'AssetsScreen', 'VerifyScreen']
+
+const SendScreenHeader = ({ navigation, route, options, back }: StackHeaderProps) => {
+  const theme = useTheme()
+
+  const [progress, setProgress] = useState(0)
+
+  useEffect(() => {
+    const routeName = getFocusedRouteNameFromRoute(route) ?? 'DestinationScreen'
+    const currentStepIndex = sendSteps.findIndex((stepName) => stepName === routeName)
+
+    setProgress(currentStepIndex / sendSteps.length)
+  }, [route])
+
+  return (
+    <BottomModalHeader>
+      <SendScreenHeaderStyled>
+        {options.headerLeft ? options.headerLeft({}) : <BackButton onPress={() => navigation.goBack()} />}
+        <ProgressBar
+          progress={progress}
+          color={theme.global.accent}
+          unfilledColor={theme.border.secondary}
+          borderWidth={0}
+          height={9}
+          width={120}
+        />
+        {options.headerRight ? (
+          options.headerRight({})
+        ) : (
+          <ContinueButton onPress={() => navigation.navigate('OriginScreen')} />
+        )}
+      </SendScreenHeaderStyled>
+    </BottomModalHeader>
+  )
+}
+
+export default SendScreenHeader
+
+export const BackButton = (props: PressableProps) => {
```

Comment:
This could probably be extracted and reused in other places.

Reply:
Also, are we ready to start DRYing the buttons code yet, or are the designs going to evolve you think? For example, the "Continue" button could possibly be the same component as the "Scan" and "Paste" ones, or the "New address" 

<img width="91" alt="image" src="https://github.com/alephium/mobile-wallet/assets/1579899/9fb9d371-6ae1-4b75-84ff-f40f11d266aa">

<img width="161" alt="image" src="https://github.com/alephium/mobile-wallet/assets/1579899/e3571c0d-a9db-4213-b3d2-0fb85633943a">

<img width="95" alt="image" src="https://github.com/alephium/mobile-wallet/assets/1579899/bd7e0799-a9ff-47d2-9031-6f7ebe0887b5">

<img width="174" alt="image" src="https://github.com/alephium/mobile-wallet/assets/1579899/e7a39ab8-8b3f-4d3a-8463-590545e99eee">

but probably a different component than

<img width="279" alt="image" src="https://github.com/alephium/mobile-wallet/assets/1579899/3ae989ed-2742-49ce-9357-2e957e368ee9">

<img width="120" alt="image" src="https://github.com/alephium/mobile-wallet/assets/1579899/5ff14436-dbbb-4b7f-8b1d-f0cbc3186cf4">

or maybe they could all be unified, but our `Buttons` component needs a lot of cleanup 😅 

I was kind waiting for more stable designs before proceeding with this, to avoid too early optimization.

Reply:
🏓  @mvaivre

Reply:
I'd wait and dry things up when cleaning up the UI before release!